src/main/java/org/apache/commons/cli/help/TextHelpWriter.java

```diff
@@ -0,0 +1,386 @@
+    public static final int DEFAULT_INDENT = 3;
+
+    /** Number of space characters before a list continuation line */
+    public static final int DEFAULT_LIST_INDENT = 7;
+
+    /** Defines the TextStyle for paragraph, and associated output formats. */
+    private final TextStyle.Builder styleBuilder;
+
+    /**
+     * Construct from an output.
+     * @param output the Appendable to write the output to.
+     */
+    public TextHelpWriter(final Appendable output) {
+        super(output);
+        styleBuilder = new TextStyle.Builder().setMaxWidth(DEFAULT_WIDTH)
+                .setLeftPad(DEFAULT_LEFT_PAD).setIndent(DEFAULT_INDENT);
+    }
+
+    /**
+     * Sets the maximum width for the output.
+     * @param maxWidth the maximum width for the output.
+     */
+    public void setMaxWidth(final int maxWidth) {
+        styleBuilder.setMaxWidth(maxWidth);
+    }
+
+    /**
+     * Gets the maximum width for the output
+     * @return the maximum width for the output.
+     */
+    public int getMaxWidth() {
+        return styleBuilder.getMaxWidth();
+    }
+
+    /**
+     * Sets the left padding: the number of characters from the left edge to start output.
+     * @param leftPad the left padding.
+     */
+    public void setLeftPad(final int leftPad) {
+        styleBuilder.setLeftPad(leftPad);
+    }
+
+    /**
+     * Returns the left padding for the output.
+     * @return The left padding for the output.
+     */
+    public int getLeftPad() {
+        return styleBuilder.getLeftPad();
+    }
+
+    /**
+     * Sets the indent for the output.
+     * @param indent the indent used for paragraphs.
+     */
+    public void setIndent(final int indent) {
+        styleBuilder.setIndent(indent);
+    }
+
+    /**
+     * Gets the indent for the output.
+     * @return the indent ofr the page.
+     */
+    public int getIndent() {
+        return styleBuilder.getIndent();
+    }
+
+    /**
+     * Gets the style builder used to format text that is not otherwise formatted.
+     * @return The style builder used to format text that is not otherwise formatted.
+     */
+    public TextStyle.Builder getStyleBuilder() {
+        return styleBuilder;
+    }
+
+    /**
+     * Print a queue of text.
+     * @param queue the queue of text to print.
+     * @throws IOException on output error.
+     */
+    private void printQueue(final Queue<String> queue) throws IOException {
+        for (String s : queue) {
+            output.append(format("%s%n", Util.rtrim(s)));
+        }
+    }
+
+    @Override
+    public void writeTitle(final CharSequence title) throws IOException {
+        if (!Util.isEmpty(title)) {
+            TextStyle style = styleBuilder.get();
+            Queue<String> queue = makeColumnQueue(title, style);
+            queue.add(Util.createPadding(style.getLeftPad()) + Util.filledString(Math.min(title.length(), style.getMaxWidth()), '#'));
+            queue.add("");
```

Comment:
```java
queue.add("");
```
This looks odd. Why are we adding an empty string? What does that do?

Reply:
Hm, this is non obvious and since this is done at least three times, I would refactor these calls into a private method; but what would it be called? `endElement(Queue)`?